File: PythonClient/diffie_hellman/diffie_hellman_test.py
from binascii import hexlify
from diffie_hellman import DiffieHellman


p = 11
g = 2

d1 = DiffieHellman(generator=g, prime=p, key_length=3)
# d2 = DiffieHellman(generator=g, prime=p, key_length=3)
# print(d1.public_key)
# print(d2.public_key)
# d1.gen_key(d2.public_key)
# d2.gen_key(d1.public_key)
d1.gen_key(2)

print('Public key: ' + str(d1.public_key))
print('Private key: ' + str(d1.private_key))
print(d1.get_key())

# d1.show_params()
# d1.show_results()
# d2.show_params()
# d2.show_results()

# if d1.get_key() == d2.get_key():
#     print('Shared keys match')
#     print('Key: ', hexlify(bytes(d1.key, 'ascii')))
#     print(d1.get_key())
# else:
#     print("Shared secrets didn't match!")
#     print("Shared secret A: ", d1.gen_secret(d2.public_key))
#     print("Shared secret B: ", d2.gen_secret(d1.public_key))

File: WebClient/app/diffie_hellman/diffie_hellman_test.js
import DiffieHellman from './diffie_hellman';


const p = 11,
      g = 2,
      keyLength = 3;

let d1 = new DiffieHellman(g, p, keyLength);
// let d2 = new DiffieHellman(g, p, keyLength);

console.log('Public keys:');
console.log(d1.getPublicKey());
// console.log(d2.getPublicKey());

// d1.genKey(d2.getPublicKey());
// d2.genKey(d1.getPublicKey());
d1.genKey(2);

console.log('Private keys:');
console.log(d1.privateKey);
// console.log(d2.privateKey);

console.log(d1.getKey());
// console.log(d2.getKey());
console.log(d1.sharedSecret);

console.log('-------');
console.log(d1.genPublicKey());
console.log(d1.sharedSecret);
